ci/ray_ci/automation/docker_tags_lib.py

```diff
@@ -97,6 +99,38 @@ def list_image_tag_suffixes(
     return tag_suffixes
 
 
+def pull_image(image_name: str) -> None:
+    """
+    Pull image from Docker Hub.
+    """
+    client = docker.from_env()
+    client.images.pull(image_name)
+
+
+def get_ray_commit(image_name: str) -> str:
```

Comment:
after pull and run, could you also delete the image, so that the machine is not left with a lot of images?

Reply:
maybe keep one image for shared layer caching for performance.

the machine might not have enough space to save all the images locally.

Reply:
I actually move `pull_image` logic out of this method and do it in the binary instead along with `remove_image`. I think this method should only run the container to grab the output (commit hash)

Reply:
^ logic to keep 1 image for caching is done also (in the binary file)